pkg/util/push/otel.go

```diff
@@ -187,14 +187,16 @@ func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
 
 	metadataLength := 0
 	for i := 0; i < resourceMetricsSlice.Len(); i++ {
-		metadataLength += resourceMetricsSlice.At(i).ScopeMetrics().Len()
+		scopeMetricsSlice := resourceMetricsSlice.At(i).ScopeMetrics()
+		for j := 0; j < scopeMetricsSlice.Len(); j++ {
+			metadataLength += scopeMetricsSlice.At(j).Metrics().Len()
+		}
 	}
+
 	var metadata = make([]*mimirpb.MetricMetadata, metadataLength)
 	var metaDataPos = 0
```

Comment:
Why do we need `metaDataPos` at all? Can't we just initialise `metadata`  capacity (instead of length) and use `append()` below? It makes me uncomfortable appending to `metadata` by index.

Reply:
Done, thanks for the suggestion.